### PR TITLE
docs(actionbutton): migrate docs to storybook

### PR DESCRIPTION
**Pull request by aramos-adobe**
## Description

Migrating `actionbutton` docs to storybook. 

Added missing variants:

- Static White & Black
- Selected 
- Emphasis
- Sizing
- Disabled

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [x] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] I have tested these changes in Windows High Contrast mode.
- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.
- [x] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/actionbutton/stories/actionbutton.stories.js [hunk 1]
```diff
@@ -110,6 +142,45 @@ Quiet.parameters = {
...
+StaticBlackDefault.tags = ["!dev"];
+StaticBlackDefault.args = {
+	label: "Edit",
+	iconName: "Edit",
+	staticColor: "black",
+};
+
+StaticBlackDefault.parameters = {
+	chromatic: { disableSnapshot: true },
+};
```
[the remark this answers is not in the document]
Comment (marissahuysentruyt) [reply components/actionbutton/stories/actionbutton.stories.js:167]: Sorry- I know I just keep replying to my own comments 🤦‍♀️

I saw there's static white (quiet) and static black (quiet) variants on [the docs site](https://opensource.adobe.com/spectrum-css/actionbutton.html). Should we include those on the docs page as well? 

Screenshot 2024-09-11 at 2 04 17 PM

File: components/actionbutton/stories/actionbutton.stories.js [hunk 2]
```diff
@@ -110,6 +142,45 @@ Quiet.parameters = {
...
+
+export const Sizing = (args, context) => Sizes({
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/actionbutton.stories.js:169]: Maybe we could use some of the descriptions from [the guidance site](https://spectrum.adobe.com/page/action-button/)? That page has a nice description about each of the stories we have.

I definitely think we should include something about the label, since it's important for accessibility.

File: components/actionbutton/stories/actionbutton.stories.js [hunk 3]
```diff
@@ -83,25 +85,55 @@ export default {
 
 export const Default = ActionButtonGroup.bind({});
 Default.args = {
-	label: "More",
+	label: "Edit",
+	iconName: "Edit",
 };
 
 export const Emphasized = ActionButtons.bind({});
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/actionbutton.stories.js:92]: The docs site has an "emphasized quiet" variant. I don't think I am really missing it- it's kind of weird to have a quiet variant, but then it be "emphasized." Maybe just a blurb in the quiet story docs? Something like "A quiet action button that is selected can also have emphasized styling." I don't know, just a thought. If we can figure out like, a class name that gets added or removed, we can always note that in the description too!

File: components/actionbutton/stories/template.js [hunk 4]
```diff
@@ -128,3 +128,41 @@ export const Template = ({
...
+    })}
+    ${Template({
+      ...args,
+      hideLabel: true,
+      iconName: iconName ?? "Edit",
+    })}
+  `,
+});
+
+export const TreatmentTemplate = (args) => Container({
```
[the remark this answers is not in the document]
Comment (marissahuysentruyt) [reply components/actionbutton/stories/template.js:158]: Could we somehow add extra `Container()`s to the `TreatmentTemplate` to simulate the headings like the docs site has. Checkbox does something like this: https://github.com/adobe/spectrum-css/blob/ac5cf82115376b95fce3f0bf8f12d8d59befa3fb/components/checkbox/stories/template.js#L130

Screenshot 2024-09-11 at 2 21 01 PM

I'm not entirely sure how to work it into your `map` function right now. Maybe something like this? 

```
export const TreatmentTemplate = (args, context) => Container({
	withBorder: false,
	direction: "column",
	wrapperStyles: {
		rowGap: "12px",
	},
	content: html`${[true, false].map((treatment) => Container({
		withBorder: false, 
		heading: treatment ? "Selected" : "Not selected", 
		content: 
		ActionButtonsWithIconOptions({ 
		...args, 
		treatment,
		isSelected: treatment }, context),
		})
	)}`,
},context);
```

**Review by marissahuysentruyt — CHANGES_REQUESTED** [review]
@aramos-adobe This is coming along nicely! For the most part, this is a first pass for me, mainly focused on the docs migrations. I found a couple places where maybe we can move over usage notes/migration notes, and then I had some questions for you. I also tried to share our latest approaches.

I'd like to go through once more, and compare your changes to the S2 action button changes ([Melissa's branch](https://github.com/adobe/spectrum-css/blob/fa9418c87ce4690a9fccc9649c8ef9001fcc1e5c/components/actionbutton/stories/actionbutton.stories.js) and then the [PR](https://github.com/adobe/spectrum-css/pull/2669/files)) and make sure anything relevant got brought over.

Way to jump in on this component- it's pretty complicated and used in a bunch of places!!

**Review by marissahuysentruyt — COMMENTED** [review]
@aramos-adobe I went through and compared this PR to the [S2 action button migration](https://github.com/adobe/spectrum-css/pull/2669/files), trying to see if there was anything we might want to grab from there. I honestly didn't see much!  I left a note about some hidden controls for the focus, hover, and active states. 

And then the only other change from that S2 branch that I'm unsure about is a [conditional to the `isEmphasized`](https://github.com/adobe/spectrum-css/blob/fa9418c87ce4690a9fccc9649c8ef9001fcc1e5c/components/actionbutton/stories/actionbutton.stories.js#L58) arg. If a user selects one of the static color variants, the `isEmphasized` control doesn't show in that case. I'm wondering if we should actually keep things the way they are, and not bring that over. The static colors are supposed to overwrite any of our other styles, including emphasized. Maybe it's handy for users to see that when they have an emphasized action button, but they also want it to be a static color variant, the styles are going to be the static color styles. I'm just thinking out loud at this point!

Last thing I promise! In the [action button test file](https://github.com/adobe/spectrum-css/blob/ac5cf82115376b95fce3f0bf8f12d8d59befa3fb/components/actionbutton/stories/actionbutton.test.js#L49), could we add: 
```
{
			testHeading: "Quiet + emphasized",
			isEmphasized: true,
			isQuiet: true,
		},
```
It's showcased on the docs site, so we should probably cover it in the VRT tests. 🤷‍♀️ I know you're already connecting with Josh, but I'm also to pair on anything if you need! 

Great job on this 🎉 ✨

File: components/actionbutton/stories/template.js [hunk 5]
```diff
@@ -128,3 +128,75 @@ export const Template = ({
...
+	direction: "row",
+	wrapperStyles: {
+		rowGap: "12px",
+	},
+	content: html`${[
+		{ isSelected: false, isDisabled: false, heading: "Default"}, 
+		{ isSelected: true, isDisabled: false, heading: "Selected" },
+		{ isSelected: false, isDisabled: true, heading: "Disabled" },
+		{ isSelected: true, isDisabled: true, heading: "Selected + disabled" }
+	].map(({isSelected, isDisabled, heading }) => Container({
```
Comment (jawinn) [inline components/actionbutton/stories/template.js:193]: Nice use of `map()` here to keep this template compact. 👍

File: components/actionbutton/stories/template.js [hunk 6]
```diff
@@ -128,3 +128,75 @@ export const Template = ({
...
+      hasPopup: "true",
+    }, context )}
+	${Template({
+      ...args,
+      hideLabel: true,
+			isQuiet: true,
+      hasPopup: "true",
+    }, context )}
+	`
+});
```
Comment (jawinn) [inline components/actionbutton/stories/template.js:180]: The indentation needs some adjustment in this function.

File: components/actionbutton/stories/template.js [hunk 7]
```diff
@@ -128,3 +128,75 @@ export const Template = ({
...
+	wrapperStyles: {
+		columnGap: "12px",
+	},
+	content: html`
+    ${Template({
+      ...args,
+    }, context )}
+    ${Template({
+      ...args,
+    }, context )}
```
Comment (jawinn) [inline components/actionbutton/stories/template.js:146]: The first two are the same here. I think the second is supposed to exclude the icon?

File: components/actionbutton/stories/actionbutton.test.js [hunk 8]
```diff
@@ -7,17 +7,15 @@ export const ActionButtons = (args, context) => {
...
 			hideLabel: true,
 		}, context)}
 		${Template({
 			...args,
-			hasPopup: "menu",
-			label: "Has pop-up",
-			iconName: undefined,
+			hasPopup: "true",
+			hideLabel: true,
 		}, context)}
```
Comment (jawinn) [inline components/actionbutton/stories/actionbutton.test.js:19]: Do you think it's worth adding a fifth action button here that displays like the original, with label text + the hold icon? Covering the hold icon for both scenarios could be useful.

File: components/actionbutton/stories/actionbutton.stories.js [hunk 9]
```diff
@@ -110,6 +140,82 @@ Quiet.parameters = {
...
 
+/**
+ * An action button can have a hold icon (a small corner triangle). This icon indicates that holding down the action button for a short amount of time can reveal a popover menu, which can be used, for example, to switch between related actions.
+ */
+
+export const HoldIcon = IconOnlyOption.bind({});
+HoldIcon.args = ["!dev"];
+HoldIcon.parameters = {
+	chromatic: {disableSnapshot: true},
+};
```
Comment (jawinn) [inline components/actionbutton/stories/actionbutton.stories.js:151]: This needs the "!dev" tag to hide it from the sidebar.

File: components/actionbutton/stories/actionbutton.stories.js [hunk 10]
```diff
@@ -110,6 +140,82 @@ Quiet.parameters = {
...
+
+export const StaticWhiteQuiet = TreatmentTemplate.bind({});
+StaticWhiteQuiet.tags = ["!dev"];
+StaticWhiteQuiet.args = {
+	staticColor: "white",
+	isQuiet: true,
+};
+
+StaticWhiteDocs.storyName = "Static white (quiet)";
```
Comment (jawinn) [inline components/actionbutton/stories/actionbutton.stories.js:172]:
```suggestion
StaticWhiteQuiet.storyName = "Static white (quiet)";
```
It looks like a couple of the static color story names got mixed up.

File: components/actionbutton/stories/actionbutton.stories.js [hunk 11]
```diff
@@ -110,6 +140,82 @@ Quiet.parameters = {
...
+};
+
+export const StaticBlackQuiet = TreatmentTemplate.bind({});
+StaticBlackQuiet.tags = ["!dev"];
+StaticBlackQuiet.args = {
+	staticColor: "black",
+	isQuiet: true,
+};
+
+StaticWhiteDocs.storyName = "Static black (quiet)";
```
Comment (jawinn) [inline components/actionbutton/stories/actionbutton.stories.js:196]:
```suggestion
StaticBlackQuiet.storyName = "Static black (quiet)";
```

**Review by jawinn — CHANGES_REQUESTED** [review]
This is looking really good so far. It's almost there. I noticed a few things that need an update.

File: components/actionbutton/stories/template.js (hunk 6 above)
Comment (marissahuysentruyt) [reply components/actionbutton/stories/template.js:185]: I believe there's some indentation fixes needed in `ActionButtonsWithIconOptions` as well.

File: components/actionbutton/stories/template.js [hunk 12]
```diff
@@ -128,3 +128,80 @@ export const Template = ({
...
+	direction: "row",
+	wrapperStyles: {
+		rowGap: "12px",
+	},
+	content: html`${[
+		{ isSelected: false, isDisabled: false, heading: "Default"}, 
+		{ isSelected: true, isDisabled: false, heading: "Selected" },
+		{ isSelected: false, isDisabled: true, heading: "Disabled" },
+		{ isSelected: true, isDisabled: true, heading: "Selected + disabled" }
+	].map(({isSelected, isDisabled, heading }) => Container({
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/template.js:198]: Could we add the extra space in a few of these objects, just to be the same on both sides? 😆 

`{ isSelected: false, isDisabled: false, heading: "Default" },`
`.map(({ isSelected, isDisabled, heading })`

File: components/actionbutton/stories/template.js [hunk 13]
```diff
@@ -128,3 +128,80 @@ export const Template = ({
...
+      hideLabel: true,
+      hasPopup: "true",
+    }, context )}
+	${Template({
+      ...args,
+			iconName: undefined,
+      hasPopup: "true",
+    }, context )}
+  `,
+}, context );
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/template.js:163]: I'm not sure we need the `context` here, passed to `Container` (I think I did that wrong in my one comment from my last review about `context`). Just looking at your other `Container`s and the props for `Container()`, I don't think they have context passed.

File: components/actionbutton/stories/actionbutton.test.js [hunk 14]
```diff
@@ -76,5 +74,9 @@ export const ActionButtonGroup = Variants({
 		testHeading: "Disabled + selected",
 		isDisabled: true,
 		isSelected: true,
+	}, {
+		testHeading: "Quiet + emphasized",
+		isEmphasized: true,
+		isQuiet: true,
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/actionbutton.test.js:80]: I would probably move this to the `testData`, instead of `stateData`. Stepper does something similar with its `Quiet + invalid` test case.

Screenshot 2024-09-23 at 10 28 51 AM

If we move this to `testData`, then we could see the visual styles for selected, hovered, etc too. 
Screenshot 2024-09-23 at 10 31 06 AM

File: components/actionbutton/stories/actionbutton.test.js (hunk 8 above)
Comment (marissahuysentruyt) [reply components/actionbutton/stories/actionbutton.test.js:19]: @aramos-adobe I think we just need to update `ActionButtons` in the test file to have the fifth button like we have in `ActionButtonsWithIconOptions`.

Docs page ✅
Screenshot 2024-09-23 at 10 25 17 AM

Test view
Screenshot 2024-09-23 at 10 25 03 AM

File: components/actionbutton/stories/actionbutton.stories.js [hunk 15]
```diff
@@ -81,27 +84,54 @@ export default {
...
+Standard.tags = ["!dev"];
+Standard.parameters = {
+	chromatic: { disableSnapshot: true },
 };
 
-export const Emphasized = ActionButtons.bind({});
+Standard.storyName = "Default";
+
+/**
+ * By default, action buttons are not emphasized. This is optimal for when an action button is not the core part of an interface, such as in application panels, where all the visual components are monochrome in order to direct focus to the content.
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/actionbutton.stories.js:109]: @aramos-adobe wanna add anything about having [a quiet-emphasized action button,](https://github.com/adobe/spectrum-css/pull/3102#discussion_r1755276853) to cover that from the static docs site?

And similar to my comment on the quiet story, we could think about trimming some of this emphasized documentation down, focusing on the emphasized paragraph and not the default stuff, to clarify.

File: components/actionbutton/stories/actionbutton.stories.js [hunk 16]
```diff
@@ -81,27 +84,54 @@ export default {
 	},
 };
 
+// ********* DOCS ONLY ********* //
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/actionbutton.stories.js:87]: Just a nit-pick/opinion here- since we're using the `Default` story for chromatic testing (so it isn't really "docs-only"), maybe this comment should get moved beneath it.

File: components/actionbutton/stories/actionbutton.stories.js [hunk 17]
```diff
@@ -81,27 +84,54 @@ export default {
...
 	},
 };
 
-export const Quiet = ActionButtons.bind({});
+/**
+ * By default, action buttons have a visible background. This style works best in a dense array of controls where the background helps to separate action buttons from the surrounding container, or to give visibility to isolated buttons.
+
+ * Alternatively, quiet action buttons can have no visible background until they’re interacted with. This style works best when a clear layout (vertical stack, table, grid) makes it easy to parse the buttons. Too many quiet components in a small space can be hard to read.
+ */
+
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/actionbutton.stories.js:131]: Just personal preference here: I'd probably remove this "By default, ..." line from the "Quiet" docs. It could go up with the `Standard` story if you really wanted, but then we could make some minimal edits to the quiet paragraph. 


```suggestion
/**
 * Quiet action buttons have no visible background until they’re interacted with. This style works best when a clear layout (vertical stack, table, grid) makes it easy to parse the buttons. Too many quiet components in a small space can be hard to read.
 */
```

**Review by marissahuysentruyt — CHANGES_REQUESTED** [review]
Nice work on these updates Aziz! It's getting really close now. I had a few more thoughts for you, and wanted to bring up a couple of questions once more.

File: components/actionbutton/stories/template.js (hunk 6 above)
Comment (jawinn) [reply components/actionbutton/stories/template.js:184]: I'm still seeing some indentation issues [here](https://github.com/adobe/spectrum-css/blob/666230f1b7c151398d2db0ec399637f3e31b9d54/components/actionbutton/stories/template.js#L132-L184). I think it's just another case of there being mixed tabs and spaces, where converting all to tabs should help with adjusting.

File: components/actionbutton/stories/actionbutton.stories.js [hunk 18]
```diff
@@ -1,8 +1,10 @@
 import { default as IconStories } from "@spectrum-css/icon/stories/icon.stories.js";
+import { Sizes } from "@spectrum-css/preview/decorators";
 import { disableDefaultModes } from "@spectrum-css/preview/modes";
 import { isActive, isDisabled, isEmphasized, isFocused, isHovered, isQuiet, isSelected, size, staticColor } from "@spectrum-css/preview/types";
 import pkgJson from "../package.json";
 import { ActionButtonGroup, ActionButtons } from "./actionbutton.test.js";
+import { ActionButtonsWithIconOptions, TreatmentTemplate } from "./template.js";
 
 /**
  * The action button component represents an action a user can take.
```
[the remark this answers is not in the document]
Comment (jawinn) [reply components/actionbutton/stories/actionbutton.stories.js:10]: It looks like there is only one note under the old migration guide that should still be included. I would add this to the hold icon story description: "Because of the way padding is calculated, the hold icon must be placed before the workflow icon in the markup."

The rest of the migration notes would be good to move into the CHANGELOG under the appropriate versions where those notes were added (usually I do a little digging on the git blame in Github to find this).

File: components/actionbutton/stories/actionbutton.stories.js [hunk 19]
```diff
@@ -98,10 +128,30 @@ Emphasized.parameters = {
 	},
 };
 
-export const Quiet = ActionButtons.bind({});
+/**
+ * When there are too many quiet action buttons in a small space bringing emphasis to them will be effective.
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/actionbutton.stories.js:132]:
```suggestion
 * Adding the `.spectrum-ActionButton--emphasized` class to a quiet action button can be effective in calling attention.
```

File: components/actionbutton/stories/template.js (hunk 6 above)
Comment (marissahuysentruyt) [reply components/actionbutton/stories/template.js:184]: ➕ 1️⃣

**Review by marissahuysentruyt — COMMENTED** [review]
This is so close!! There's just a few more really small things to take care and this will be good to merge! 

I left a suggestion on rewording something, and I also still see the tabs vs. spacing thing Josh found. Josh also mentioned that most of the migration guide stuff (at the very [bottom of the docs page](https://opensource.adobe.com/spectrum-css/actionbutton.html)) could be put into the CHANGELOG.

File: components/actionbutton/stories/actionbutton.stories.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (marissahuysentruyt) [reply components/actionbutton/stories/actionbutton.stories.js:10]: @aramos-adobe If you could move the migration guide to the [action button CHANGELOG](https://github.com/adobe/spectrum-css/blob/main/components/actionbutton/CHANGELOG.md), that'd be great! If/when we stop supporting the static docs site, we don't want to lose all the migration guide stuff you found in the `yml` files, so that's where we've been moving them (from the yml and into the changelog).

And then if we could add Josh's suggestion above for the hold icon docs, that'd be greeaaatttt 🔥

File: components/actionbutton/stories/actionbutton.stories.js [hunk 20]
```diff
@@ -1,11 +1,21 @@
...
-import { ActionButtonGroup, ActionButtons } from "./actionbutton.test.js";
+import { ActionButtonGroup } from "./actionbutton.test.js";
+import { ActionButtonsWithIconOptions, IconOnlyOption, TreatmentTemplate } from "./template.js";
 
 /**
  * The action button component represents an action a user can take.
+ * 
+ * ## Usage notes  
+ * 
+ * For Action Buttons that only contain an icon with no label, do not include the element with the `.spectrum-ActionButton-label` class in the markup. If an icon and a label are both used, ensure that the element with the `.spectrum-ActionButton-label` class comes after the `.spectrum-Icon` element.
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/actionbutton.stories.js:14]:
```suggestion
* For action buttons that only contain an icon with no label, do not include the element with the `.spectrum-ActionButton-label` class in the markup. If an icon and a label are both used, ensure that the element with the `.spectrum-ActionButton-label` class comes after the `.spectrum-Icon` element.
```

We've been slowly correcting component names to make sure they're [sentence-case](https://spectrum.adobe.com/page/grammar-and-mechanics/#Sentence-case) and I believe this would fall under the "UI elements" bit. Changing to lowercase here matches what we've done in some other component documentation, at the very least.

File: components/actionbutton/stories/actionbutton.stories.js [hunk 21]
```diff
@@ -1,11 +1,21 @@
...
+ * 
+ * If the hold icon is used, ensure that the element with the `.spectrum-ActionButton-hold` class comes before the `.spectrum-Icon` element.
+ * 
+ * When using `.spectrum-ActionButton--staticWhite` or `.spectrum-ActionButton--staticblack`, use the `--mod-actionbutton-content-color-default` custom property to set the text color when selected.
```
Comment (marissahuysentruyt) [inline components/actionbutton/stories/actionbutton.stories.js:18]: Could you capitalize this please? `.spectrum-ActionButton--staticeBlack`. It took me a hot second to find these styles, but [static colors are capitalized](https://github.com/adobe/spectrum-css/blob/5f523698503e5683f5a769b0134e092238f7e1b0/components/actionbutton/themes/spectrum.css#L55) in their modifier class names.

File: components/search/stories/search.stories.js
Comment (marissahuysentruyt) [inline components/search/stories/search.stories.js:1]: These changes feel like maybe they were supposed to be in a different PR (maybe #3174?) Should these be dropped from this PR at all?

**Review by marissahuysentruyt — APPROVED** [review]
I think this is in good shape now! I believe everything I mentioned has been taken care, so thank you for all the dedication! 

I don't know about you, but I love a good list, so I use those checklists in the PR description and I'll check stuff off, like when I test anything in WHCM, or the bit about adding documentation, running VRT, etc. I know you ran VRTs at one point, but I don't know if we have all of the changes captured for this branch yet. This was the most recent one I could find (I think?) https://www.chromatic.com/build?appId=64762974a45b8bc5ca1705a2&number=3186, and it's missing the 5th button we added. We'll have to run those before this gets merged, but I think it's in a good spot for me to approve! 

Thanks for all your hard work!

**Review by castastrophe — APPROVED** [review]
Great job with these updates!
